build.sbt

```diff
@@ -66,12 +66,6 @@ val isScala3Setting = Def.setting {
 }
 
 def isScala3(v: Option[(Long, Long)]): Boolean = v.exists(_._1 != 2)
-val isScalaJS = Def.setting[Boolean](
-  SettingKey[Boolean]("scalaJSUseMainModuleInitializer").?.value.isDefined
-)
-val isScalaNative = Def.setting[Boolean](
-  SettingKey[String]("nativeGC").?.value.isDefined
```

Comment:
`nativeGC` is now a `TaskKey` so we can't depend on it to produce a `Setting`.

Since this was only used to set the `unmanagedSourceDirectories` below, I've moved the relevant setting down to `nativeSettings` and `jsSettings` so we don't need `isScalaNative` and `isScalaJS` anymore